duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -172,302 +199,461 @@ namespace caf {
+    if(!dirReco.failedToGet()){
+      dirBranch.lngtrk.SetX(dirReco->fRecoDirection.X());
+      dirBranch.lngtrk.SetY(dirReco->fRecoDirection.Y());
+      dirBranch.lngtrk.SetZ(dirReco->fRecoDirection.Z());
+    }
+    else{
+      std::cout << "Warning: No AngularRecoOutput found with label '" << fDirectionRecoLabelNumu << "'" << std::endl;
+    }
 
-        // The iteration order here is the same as how we filled SRGlobal, so
-        // no need to do any work to make sure they align.
-        //
-        // NB this will all go wrong if we ever support more than one MCTruth
-        // per event.
-        for(const std::vector<systtools::VarAndCVResponse>& resp: *syst_resp){
-          for(const systtools::VarAndCVResponse& it: resp){
-            // Need begin/end to convert double to float
-            sr.xsSyst_wgt.emplace_back(it.responses.begin(), it.responses.end());
-            sr.cvwgt.push_back(it.CV_response);
-            sr.total_xsSyst_cv_wgt *= it.CV_response;
-          }
-        }
+    dirReco = evt.getHandle<dune::AngularRecoOutput>(fDirectionRecoLabelNue);
+    if(!dirReco.failedToGet()){
+      dirBranch.heshw.SetX(dirReco->fRecoDirection.X());
+      dirBranch.heshw.SetY(dirReco->fRecoDirection.Y());
+      dirBranch.heshw.SetZ(dirReco->fRecoDirection.Z());
+    }
+    else{
+      std::cout << "Warning: No AngularRecoOutput found with label '" << fDirectionRecoLabelNue << "'" << std::endl;
+    }
+  }
+
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillEnergyInfo(caf::SRNeutrinoEnergyBranch &ErecBranch, const art::Event &evt) const
+  {
+    //Filling the reg CNN results
+    art::InputTag itag(fRegCNNLabel, "regcnnresult");
+    art::Handle<std::vector<cnn::RegCNNResult>> regcnn = evt.getHandle<std::vector<cnn::RegCNNResult>>(itag);
+    if(!regcnn.failedToGet() && !regcnn->empty()){
+        const std::vector<float>& cnnResults = (*regcnn)[0].fOutput;
+        ErecBranch.regcnn = cnnResults[0];
+    }
+    else{
+      std::cout << "Warning: " << itag << " does not correspond to a valid RegCNNResult product" << std::endl;
+    }
+
+    art::Handle<dune::EnergyRecoOutput> ereconue = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNueLabel);
+    art::Handle<dune::EnergyRecoOutput> ereconumu = evt.getHandle<dune::EnergyRecoOutput>(fEnergyRecoNumuLabel);
+
+
+    if(ereconue.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNueLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.calo = ereconue->fNuLorentzVector.E();
+    }
+
+    if(ereconumu.failedToGet()){
+      std::cout << "Warning: " << fEnergyRecoNumuLabel << " does not correspond to a valid EnergyRecoOutput product" << std::endl;
+    }
+    else{
+      ErecBranch.lep_calo = ereconumu->fNuLorentzVector.E();
+    }
+
```

Comment:
I'm not sure I understand the naming here.  nue goes to calo but numu goes to lep_calo?

Reply:
The naming is not ideal beacuse of different conventions in between ND and FD for the energy determination. The branch only proposes:
```c++
class SRNeutrinoEnergyBranch
  {
    private:
      static constexpr float NaN = std::numeric_limits<float>::signaling_NaN();

    public:
      float calo     = NaN;  ///< Calorimetric estimate using all hits
      float lep_calo = NaN;  ///< Lepton (longest track or largest shower) + calorimetric estimate from remaining hits
      float regcnn   = NaN;  ///< Regression CNN (assumes nue hypothesis)
  };
```

with `calo` and `lep_calo`.

The FD nomenclature for now comes from:
```
# RecoMethod is method of neutrino energy reconstruction 
    # 1 Split event into longest reco track and hadronic part, suitable for numu CC events (method 3 is used for events with no reco track)
    # 2 Split event into reco shower with highest charge and hadronic part, suitable for nue CC events (method 3 is used for events with no reco shower)
    # 3 Use charges of all hits and convert to energy, suitable for NC events, numu CC events with no reco track or nue CC events with no reco shower
```
that correspond to the nu_mu, nu_e and NC methods.
So for now I am matching method 1 to `lep_calo` and method 2 to `calo` (I think that method 2 and 3 only differ by the mass of an electron so 0.5MeV).

Reply:
I think this is worth a conversation with Jeremy as the naming is quite confusing.